libminifi/include/core/logging/alert/AlertSink.h

```diff
@@ -0,0 +1,109 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <deque>
+#include <mutex>
+#include <unordered_set>
+#include <regex>
+#include <utility>
+#include <string>
+#include <memory>
+
+#include "controllers/SSLContextService.h"
```

Comment:
This include breaks the Windows build, because `SSLContextService.h` pulls in `wincrypt.h`, which `#define`s `X509_CERT_PAIR` as `(LPCSTR) 53`, but OpenSSL/LibreSSL has its own `X509_CERT_PAIR` struct which conflicts with this.

Specifically, in `ExtendedKeyUsage.cpp`, we first `#include <openssl/x509v3.h>`, then we `#include "core/logging/LoggerConfiguration.h --> AlertSink.h --> SSLContextService.h --> wincrypt.h`, which redefines `X509_CERT_PAIR`.

I think the best solution would be to move `#include "controllers/SSLContextService.h"` to `AlertSink.cpp`, and put a forward declaration in the header instead.

Reply:
done